src/com/softwaret/kpdf/controller/publication/PublicationsControllerImpl.kt

```diff
@@ -16,8 +16,22 @@ class PublicationsControllerImpl(
     override fun obtainPublication(id: Id) =
         interactor.obtainPublicationOrNull(id)?.let {
             Response.OK(PublicationResponseBody(id, it.name, it.author.login, it.pdf.pdfBase64))
-        } ?: Response.NotFound(EmptyResponseBody)
+        } ?: Response.BadRequest(EmptyResponseBody)
 
     override fun insertPublication(publicationName: String, pdfBase64: PdfBase64, login: Login) =
-        obtainPublication(interactor.insertPublication(publicationName, pdfBase64, login))
+        obtainPublication(id = interactor.insertPublication(publicationName, pdfBase64, login))
+
+    override fun deletePublication(id: Id) =
+        ifPublicationExists(id) { interactor.deletePublication(id) }
+
+    override fun updatePublication(id: Id, pdfBase64: PdfBase64) =
+        ifPublicationExists(id) { interactor.updatePublication(id, pdfBase64) }
+
+    private fun ifPublicationExists(id: Id, action: () -> Unit) =
```

Comment:
Zmieniłbym nazwę tej funkcji aby mówiła, że zwraca jakiś `Response`,  bo obecnie nawet jak nie znajdzie publikacji o danym `id` to i tak zwraca response, a nazwa brzmi jakby funkcja miała cokolwiek robić gdy publikacja istnieje. Pierwsza myśl 
```Kotlin
override fun updatePublication(id: Id, pdfBase64: PdfBase64) =
        respondBaseOnPublicationExistence(id) { interactor.updatePublication(id, pdfBase64) }

    private fun respondBaseOnPublicationExistence(
        id: Id,
        notExistsAction: () -> Unit = {},
        existsAction: () -> Unit
    ) = if (interactor.obtainPublicationOrNull(id) == null) {
        notExistsAction()
        Response.BadRequest(EmptyResponseBody)
    } else {
        existsAction()
        Response.OK(EmptyResponseBody)
    }
```
Ale pachnie lekko over-engineeringiem

Reply:
Na ten moment nie kombinowałbym z dodawaniem tam drugiej lambdy. Ta metoda dobrze się spisuje w tym co robi.

Reply:
Oki, zastanawia mnie czy rozpoczynanie funkcji od `if`, która przykrywa `if/else` to dobry pomysł